cpmpy/transformations/normalize.py

```diff
@@ -34,3 +37,96 @@ def unravel(lst, append):
     unravel((cpm_expr,), append)
 
     return newlist
+
+
+def simplify_boolean(lst_of_expr, num_context=False):
+    """
+    removes boolean constants from all CPMpy expressions
+    only resulting boolean constant is literal 'false'
+    - list_of_expr: list of CPMpy expressions
+    """
+
+    newlist = []
+    for expr in lst_of_expr:
+
+        if isinstance(expr, bool):
+            # not sure if this should happen here or at construction time
+            expr = BoolVal(expr)
+
+        if isinstance(expr, BoolVal):
+            newlist.append(int(expr.value()) if num_context else expr)
+
+        elif isinstance(expr, Operator):
+            args = simplify_boolean(expr.args, num_context=not expr.is_bool())
```

Comment:
Comparisons are Boolean, but their arguments can be int? So I guess we should upcast those too?

Reply:
I'm missing something here... args is empty for comparisons?

Reply:
Comparisons are not operators :)

Reply:
Maybe add a `TODO: check code in typecheck merge request. There may be an operator -> argument type array there`.

Reply:
currently all operators with bool output type have bool input types. I'm not sure where this should be documented... we also don't really check it here, it just influences whether '1' or True is returned.

Reply:
So for now we simply leave the check as is?